packages/rocketchat-lib/server/methods/setEmail.coffee

```diff
@@ -0,0 +1,27 @@
+Meteor.methods
+	setEmail: (email) ->
+		if not Meteor.userId()
+			throw new Meteor.Error('invalid-user', "[methods] setEmail -> Invalid user")
+
+		user = Meteor.user()
+
+		if not RocketChat.settings.get("Accounts_AllowEmailChange")
+			throw new Meteor.Error(403, "[methods] setEmail -> E-mail change not allowed")
+
+		if user.emails?[0]?.address is email
+			return email
+
+		emailValidation = /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/
+		if not emailValidation.test email
```

Comment:
Are you talking about once on method and another on function? Usually on functions we don't return an error, just false for end of execution. Functions are intended to be run by methods or by api calls, AFAICT. Let's discuss with the group if that's the real intention...